---
title: "miRNA data Analysis"
author: "Vijender Singh, CBC, UCHC"
date: "`2024-05-01"
output:
  pdf_document:
email: vijender.singh@uconn.edu
---
```{r setup, include=TRUE}
rm(list=ls())
gc()

knitr::opts_chunk$set(echo = TRUE, warning = FALSE)

# Load Packages
suppressPackageStartupMessages(library("tidyverse",quietly=TRUE))
suppressPackageStartupMessages(library("genefilter",quietly=TRUE))
suppressPackageStartupMessages(library("grDevices",quietly=TRUE))
suppressPackageStartupMessages(library("ggrepel",quietly=TRUE))
suppressPackageStartupMessages(library("pheatmap",quietly=TRUE))
suppressPackageStartupMessages(library("RColorBrewer",quietly=TRUE))
suppressPackageStartupMessages(library("gplots",quietly=TRUE))
suppressPackageStartupMessages(library("RColorBrewer", quietly=TRUE))
suppressPackageStartupMessages(library("ggplot2", quietly=TRUE))
suppressPackageStartupMessages(library("labeling", quietly=TRUE))
suppressPackageStartupMessages(library("DESeq2",quietly=TRUE))
suppressPackageStartupMessages(library("skimr",quietly=TRUE))
suppressPackageStartupMessages(library("here",quietly=TRUE))
suppressPackageStartupMessages(library("data.table",quietly=TRUE))

#Project Directory
projectDir=setwd(here("."))
print(paste0("Project Directory :", projectDir ))
```


# RNAseq analysis : All Samples{.tabset}
## Data Import
#### Import Counts Matrix and Sample Metadata
The data is in tab-separated test (.txt) format.  Data is imported using `fred` function from `data.table` package. The sample metadat is vaialable in `sampleTable.csv` file and is read in using `read.csv` function. 
```{r dataImport, include=TRUE, results="hold"}
# Read in counts and assign geneID as rowname
countMat<-fread(file="./counts/count_matrix.txt") %>% column_to_rownames(.,var="geneID")

# Read in sampleTable and assign sample column values as rownames
sampleTable<-read.csv(file="SampleTable.csv") %>% mutate(., rowname=sample) %>% column_to_rownames(., var="rowname")
sampleTable$Condition<-as.factor(sampleTable$Condition)

#Setting up reference level for comparisons
ref_level="Old"

# Print out Data info
cat("\n Count matrix Stats: \n Features(genes) : ", nrow(counts), "\n Number of samples : ", ncol(counts))
cat("\n \n Sample Meatadata\n")
sampleTable

```


#### Filtering
In this step we will filter out genes with very low counts across the samples.  DESeq2 performs independent filtering but reducing the dimension of input dataset can speed up the process in large datasets.

```{r countFiltering, include=TRUE, results="hold"}
keep <- rowSums(countMat) >= 10
countMat<-countMat[keep,]

cat("\n Number of features after filteration : ", nrow(countMat))
```


## Counts QC

In this step we will perform few QC steps to check data completeness and will also try to identify if any sample is behaving odd from the raw counts perspective.

##### QC1. Sample checks
Test that all the input samples in sampleTable have their corresponding counts in the `counts` object.
```{r sampletable2, include=TRUE, results="hold"}

if (sum(colnames(countMat) %in% sampleTable$sample) == length(sampleTable$sample)){
  message("Good News !!! Samples in count matrix matches with that of in sampleTable")
}

cat("\n\n Matching order of samples between counts and sampleTable \n")
countMat<-countMat[sampleTable$sample]

cat("\n Is the sample orders identical between counts matrix and sampletable \n")
identical(colnames(countMat),sampleTable$sample)

cat("\n\n Convert metadata in categorical data whereever applicable\n")
sampleTable$Condition<-as.factor(sampleTable$Condition)

str(sampleTable)

```

##### QC2. QC of counts distribution per sample.

```{r QC2 ,echo=FALSE, include=TRUE, warning=FALSE, results="asis",fig.height = 6, fig.width = 8, fig.align = "center"}
df.m <- reshape2::melt(countMat, id.vars =NULL)
plot_count_violin<- ggplot(df.m,aes(factor(variable),log10(value),fill=factor(variable))) +
  geom_violin() +
  theme(axis.text.x = element_text(angle=90,vjust=0.5,hjust=1))

cat(paste0("####  Sample: Count distribution Violin plot  {.tabset} \n")) 
cat('\n##### **Violin Count Plot** \n')
suppressWarnings(print(plot_count_violin))
cat('\n\n')
```


Quality check of samples to identify outliers or odd behaving samples.
```{r  include=TRUE, warning=FALSE}
df_qc<-skim(countMat)
head(df_qc)
```
Look at the distribution of mean and std.deviation of counts across all samples.
```{r QC1 ,echo=FALSE, include=TRUE, warning=FALSE, results="asis",fig.height = 6, fig.width = 8, fig.align = "center"}
mean_sd_plot <- ggplot(df_qc,aes(numeric.mean,numeric.sd, col=sampleTable$genotype_condition))+
  geom_point(size=3, alpha=0.6)+
  geom_text_repel(label=df_qc$skim_variable)
cat(paste0("####  Sample Counts : Mean  and Standard Deviation  {.tabset} \n")) 
cat('\n##### **Counts Mean and Standard deviation plot** \n')
suppressWarnings(print(mean_sd_plot))
cat('\n\n') 
```

## DESeq2

##### Create DESeq2 Object
In this analysis `control` samples are used as refrence condition for statistical analysis.
```{r  DESeq2_obj,include=TRUE, warning=FALSE} 
# reassuring that order of samples in rows of sampleTable are identical with the sample order in count matrix
countMat<-countMat[sampleTable$sample]

dds <- DESeqDataSetFromMatrix(countData = as.matrix(countMat),
                              colData=sampleTable,
                              design = ~Condition)

# Setting "control" as baseline or reference
dds$Condition<-relevel(dds$Condition, ref=ref_level)

dds
```
##### Executing DESeq2
```{r  DESeq2_funcn,include=TRUE, warning=FALSE,  results="hold"} 

dds<-DESeq(dds)

sizeFactors(dds)
```

Accessing results 
```{r  DESeq2_reslts,include=TRUE, warning=FALSE} 
resultsNames(dds)

## Access and saving normalised counts
normcounts<-as.data.frame(counts(dds,normalized =TRUE))
write.csv(normcounts,file="NormalisedCounts.csv")
```

## Sample QC
Sample to sample Distance Matrix
```{r Sample2Sample_DistMatrix, include=TRUE, warning=FALSE}

rld <- rlogTransformation(dds, blind=T)
vsd <- varianceStabilizingTransformation(dds, blind=T)

sampleDists <- dist(t(assay(rld)))
sampleDistMatrix <- as.matrix(sampleDists)

sampleDistMatrix_meta<-merge(sampleDistMatrix,sampleTable, by=0, sort=FALSE) %>% column_to_rownames(., var="Row.names")

head(sampleDistMatrix_meta)
```

```{r PlotSample2SampleDistMtx ,echo=FALSE, include=TRUE, warning=FALSE, results="asis",fig.height = 4, fig.width = 8, fig.align = "center"}

ann_df<-sampleDistMatrix_meta[c(10)]
#str(sampleTable)
#control PNA_10b PNA_21 PNA10+21
ann_colors <- list( condition = c(
              "Old" ="coral1",
              "Young" = "steelblue1"))

heat_map <- pheatmap(sampleDistMatrix,
                     labels_row = rownames(sampleDistMatrix),
                     labels_col = rownames(sampleDistMatrix),
                     annotation_row = ann_df, 
                     annotation_col = ann_df, 
                     annotation_colors = ann_colors,
                     cellwidth=14,
                     #cellheight=12
                     )

```

Principal component Analysis

Principal Component Analysis (PCA) is a widely used statistical method in the analysis of differential gene expression data. It helps in understanding the underlying structure of high-dimensional data, like gene expression profiles, by reducing its complexity. PCA achieves this by transforming the original variables into a new set of uncorrelated variables called principal components, which are ordered so that the first few retain most of the variation present in all of the original variables.

In the context of differential gene expression, PCA is particularly useful for several reasons:

1. Data Visualization: PCA allows for the visualization of complex gene expression data in two or three dimensions. This can help in identifying patterns, trends, or clusters in the data that might indicate similarities or differences in gene expression under various conditions or among different samples.

2. Noise Reduction: By focusing on principal components that capture the most variance, PCA can help filter out noise and reduce the dimensionality of the data, making further analysis more tractable.

3. Identification of Important Genes: The loading scores of the principal components can be used to identify genes that contribute most to the variance in the data. These genes can be candidates for further study as they might play significant roles in the biological processes or conditions being studied.

4. Batch Effect Correction: PCA can help in identifying and correcting for batch effects, which are technical non-biological differences between batches of samples that can affect gene expression levels.

5. Comparative Analysis: By comparing the PCA results of different conditions or experiments, researchers can identify shifts in the principal component space, indicating changes in gene expression profiles associated with different biological states or treatments.

```{r PCA_plots_cal, include=TRUE, warning=FALSE}
rv <- rowVars(assay(rld))
select <- order(rv, decreasing=T)[seq_len(min(100,length(rv)))]
pc <- prcomp(t(assay(vsd)[select,]))
scores <- data.frame(pc$x)
scores<-cbind(scores,sampleTable)
summary(pc)
pc1prcnt=as.character(as.integer(summary(pc)$importance[2,1]*100))
pc2prcnt=as.character(as.integer(summary(pc)$importance[2,2]*100))
pc3prcnt=as.character(as.integer(summary(pc)$importance[2,3]*100))
pc4prcnt=as.character(as.integer(summary(pc)$importance[2,4]*100))
pc5prcnt=as.character(as.integer(summary(pc)$importance[2,5]*100))
pc6prcnt=as.character(as.integer(summary(pc)$importance[2,6]*100))

names<-rownames(scores)

pc12 <-ggplot(scores, aes(x = PC1, y = PC2, label=sample,col = (factor(Condition))))+
  geom_point(size = 5)+
  geom_text_repel(label=names,col="black")+
  ggtitle("Principal Components")+
  xlab(paste0("PC1 (",pc1prcnt,"%)")) +
  ylab(paste0("PC2 (",pc2prcnt,"%)"))

pc34 <-ggplot(scores, aes(x = PC3, y = PC4, label=sample,col = (factor(Condition))))+
  geom_point(size = 5)+
  geom_text_repel(label=names,col="black")+
  ggtitle("Principal Components")+
  xlab(paste0("PC3 (",pc3prcnt,"%)")) +
  ylab(paste0("PC4 (",pc4prcnt,"%)"))

pc56 <-ggplot(scores, aes(x = PC5, y = PC6, label=sample,col = (factor(Condition))))+
  geom_point(size = 5)+
  geom_text_repel(label=names,col="black")+
  ggtitle("Principal Components")+
  xlab(paste0("PC5 (",pc5prcnt,"%)")) +
  ylab(paste0("PC6 (",pc6prcnt,"%)"))
```

```{r PCA_plots_plot, echo=FALSE, include=TRUE, warning=FALSE, results="asis",fig.height = 6, fig.width = 8, fig.align = "center"}

write.csv(as.data.frame(pc$rotation), file="PrincipalComponentLoadingGenes.csv")

plts_PCA<-list(pc12,pc34,pc56)
labels_pcs<-c("PC12","PC34","PC56")
cat("\n### PCA Plots  {.tabset} \n") 
for(i in seq(1,3)){
  cat('#### Plot',labels_pcs[i],'   \n')
  cat('\n')
  suppressWarnings(print(plts_PCA[[i]]))
  cat('\n\n')
}
```



```{r Save results, include=TRUE, warning=FALSE, results="asis"}

res<-as.data.frame(results(dds, name = "Condition_Young_vs_Old"))

write.csv(res, file="AllDetectedFeatures_DE.csv")

resOrdered <- res[order(res$padj),]
resOrdered

write.csv(resOrdered, file="AllDetectedFeatures_DE_padjSorted.csv")

resSign <- res[res$pvalue<0.05 & abs(res$log2FoldChange)>1,]
write.csv(resSign, file="Pval_0.05_log2FC_1_significant_features.csv")

```





